### PR TITLE
Add initial support for stm32

### PR DESCRIPTION
**Pull request by thom24**
This pull request adds an initial support for stm32 boards.

For now only the stm32mp1 subtarget was added.
The first defined device is STM32MP135F-DK.

### CODE REVIEW

**Comment by hauke** [comment]
The CI is failing like this:
```
WARNING: Makefile 'package/utils/audit/Makefile' has a dependency on 'libev', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/busybox/Makefile' has a build dependency on 'libpam', which does not exist
WARNING: Makefile 'package/boot/kexec-tools/Makefile' has a dependency on 'liblzma', which does not exist
WARNING: Makefile 'package/network/services/lldpd/Makefile' has a dependency on 'libnetsnmp', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a dependency on 'libpam', which does not exist
WARNING: Makefile 'package/utils/policycoreutils/Makefile' has a build dependency on 'libpam', which does not exist
make[2]: Entering directory '/__w/openwrt/openwrt/openwrt/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/__w/openwrt/openwrt/openwrt/scripts/config'
Checking 'python3-cryptography'... ok.
Checking 'python3-pyelftools'... ok.
Checking 'dtc'... time: target/linux/prereq#0.14#0.07#0.19
failed.

optee-os: Please install the device-tree-compiler package

make[3]: *** [/__w/openwrt/openwrt/openwrt/include/prereq.mk:9: prereq] Error 1
Checking 'python3-dev'... ok.
Checking 'python3-setuptools'... ok.
Checking 'swig'... ok.
Package prerequisite check failed.
make[2]: *** [Makefile:86: tmp/.prereq_packages] Error 1
make[1]: *** [/__w/openwrt/openwrt/openwrt/include/toplevel.mk:226: prereq] Error 2
make: *** [/__w/openwrt/openwrt/openwrt/include/toplevel.mk:246: tools/quilt/compile] Error 2
Error: Process completed with exit code 2.
```
Could you have a look at this please.

File: package/firmware/murata-firmware/Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+PKG_VERSION:=2020.09.14
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/murata-wireless/cyw-fmac-fw.git
+PKG_MIRROR_HASH:=c2fc14802860f923d7bea36a709efeeffce4637a57084d78133e080b3dd8a87c
+PKG_SOURCE_VERSION:=52174a18134c7ef4a674ecd9fb68fc6e2bced969
+
+PKG_MAINTAINER:=Thomas Richard <thomas.richard@bootlin.com>
+PKG_LICENSE_FILES:=LICENCE
```
Comment (hauke) [inline package/firmware/murata-firmware/Makefile:20]: The license looks ok:
> Subject to the terms and conditions of this
Agreement, Cypress Semiconductor Corporation ("Cypress") grants you a
non-exclusive, non-transferable license under its copyright rights in the
Software to reproduce and distribute the Software in object code form only,
solely for use in connection with Cypress integrated circuit products
("Purpose").

The firmware situation around the cypress chips is a mess. It would be so much easier if this would be in linux-firmware. 
@thom24 This is not your fault and you can probably also not improve this much.

File: package/firmware/murata-firmware/Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		$(PKG_BUILD_DIR)/cyfmac43430-sdio.1DX.clm_blob \
+		$(1)/lib/firmware/murata/
+	$(INSTALL_DIR) $(1)/lib/firmware/brcm
+	$(LN) \
+		../murata/cyfmac43430-sdio.bin \
+		$(1)/lib/firmware/brcm/brcmfmac43430-sdio.bin
+	$(LN) \
+		../murata/cyfmac43430-sdio.1DX.clm_blob \
+		$(1)/lib/firmware/brcm/brcmfmac43430-sdio.clm_blob
+endef
```
Comment (hauke) [inline package/firmware/murata-firmware/Makefile:54]: What is the difference to the `cypress-firmware-43430-sdio` package?
It also installs `/lib/firmware/brcm/brcmfmac43430-sdio.bin` and will conflict with this package. 

You should define `CONFLICTS:=cypress-firmware-43430-sdio`

File: target/linux/stm32/base-files/lib/preinit/82_brcmfmac_fix_fw_names [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+. /lib/functions/system.sh
+
+brcmfmac_fix_fw_names() {
+	local fw="/lib/firmware/brcm/brcmfmac43430-sdio"
+	local board="$(board_name)"
+
+	for ext in bin clm_blob txt
+	do
+		[ ! -f "${fw}.${board}.${ext}" ] && [ -f "${fw}.${ext}" ] && \
+			ln -s "${fw##*/}.${ext}" "${fw}.${board}.${ext}"
```
Comment (hauke) [inline target/linux/stm32/base-files/lib/preinit/82_brcmfmac_fix_fw_names:11]: Why is this needed?
Why don't you directly create the symlinks at build time?

File: target/linux/stm32/base-files/lib/preinit/83_switch_to_inband_interrupt_for_wifi_sdio [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+. /lib/functions/system.sh
+
+switch_to_inband_interrupt_for_wifi_sdio() {
+	[ -e /lib/firmware/brcm/brcmfmac43430-sdio.txt ] && \
+		sed -i -E -e 's/^muxenab=0x10/#muxenab=0x10/' "$(readlink -f /lib/firmware/brcm/brcmfmac43430-sdio.txt)"
```
Comment (hauke) [inline target/linux/stm32/base-files/lib/preinit/83_switch_to_inband_interrupt_for_wifi_sdio:6]: I would prefer to have this at build time.
I think every board should have its unique nvram file.

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+CONFIG_ARM_SCMI_TRANSPORT_SMC=y
+# CONFIG_ARM_SCMI_TRANSPORT_SMC_ATOMIC_ENABLE is not set
+CONFIG_ARM_SMCCC_SOC_ID=y
+CONFIG_ARM_SMC_WATCHDOG=y
+# CONFIG_ARM_SMMU is not set
+CONFIG_ARM_THUMB=y
+CONFIG_ARM_THUMBEE=y
+CONFIG_ARM_UNWIND=y
+CONFIG_ARM_VIRT_EXT=y
+CONFIG_ASN1=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:66]: Why do you need ASN1 support?

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 6]
```diff
@@ -0,0 +1,604 @@
...
+CONFIG_CRYPTO_SIG2=y
+CONFIG_CURRENT_POINTER_IN_TPIDRURO=y
+CONFIG_DCACHE_WORD_ACCESS=y
+CONFIG_DEBUG_ALIGN_RODATA=y
+CONFIG_DEBUG_BUGVERBOSE=y
+CONFIG_DEBUG_INFO=y
+CONFIG_DEBUG_LL_INCLUDE="mach/debug-macro.S"
+CONFIG_DEBUG_MISC=y
+CONFIG_DEBUG_PREEMPT=y
+CONFIG_DEBUG_SECTION_MISMATCH=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:180]: Please remove this.

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 7]
```diff
@@ -0,0 +1,604 @@
...
+CONFIG_ELF_CORE=y
+CONFIG_EXCLUSIVE_SYSTEM_RAM=y
+CONFIG_EXT4_FS=y
+CONFIG_EXTCON=y
+CONFIG_EXTCON_USB_GPIO=y
+CONFIG_FAILOVER=y
+CONFIG_FHANDLE=y
+CONFIG_FIXED_PHY=y
+CONFIG_FIX_EARLYCON_MEM=y
+# CONFIG_FORTIFY_SOURCE is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:227]: Do not unset `CONFIG_FORTIFY_SOURCE`

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 8]
```diff
@@ -0,0 +1,604 @@
...
+CONFIG_GENERIC_TIME_VSYSCALL=y
+CONFIG_GENERIC_VDSO_32=y
+CONFIG_GLOB=y
+CONFIG_GPIOLIB_IRQCHIP=y
+CONFIG_GPIO_CDEV=y
+CONFIG_GPIO_GENERIC=y
+CONFIG_GPIO_GENERIC_PLATFORM=y
+CONFIG_GPIO_SYSCON=y
+# CONFIG_GPIO_SYSFS is not set
+# CONFIG_HARDENED_USERCOPY is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:273]: Do not remove `CONFIG_HARDENED_USERCOPY`.

File: target/linux/stm32/stm32mp1/config-6.6
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:1]: The configuration file should be minimal to boot up fine and have network working. All the extra features should be added as modules especially stuff which is optional on boards.
Most options should come from the `target/linux/generic/config-6.6` file and this should only contain which is specifically needed for this SoC.

File: package/firmware/murata-firmware/Makefile (hunk 1 above)
Comment (thom24) [reply package/firmware/murata-firmware/Makefile:20]: I used firmware and configs provided by ST.
It seems murata provides cypress firmware with custom CLM Blob.
There are cypress firmware and CLM Blob in linux-firmware.
But they are older than in cypress repository.

I don't really know how to deal with all of this.
Is it ok for you @hauke to use firmware, CLM blob and config from murata repositories ?
If not what do you expect ?

File: package/firmware/murata-firmware/Makefile (hunk 2 above)
Comment (thom24) [reply package/firmware/murata-firmware/Makefile:54]: Murata provides custom CLM Blob.

File: target/linux/stm32/stm32mp1/config-6.6 (hunk 5 above)
Comment (thom24) [reply target/linux/stm32/stm32mp1/config-6.6:66]: It was selected by CRYPTO_RSA.
I removed CRYPTO_RSA, so ASN1 is not set anymore.

File: target/linux/stm32/stm32mp1/config-6.6
Comment (thom24) [reply target/linux/stm32/stm32mp1/config-6.6:1]: I just did a big cleanup in the config.
If you (or any other people) have other comments on the kernel config, please let me know.

File: package/firmware/murata-firmware/Makefile (hunk 1 above)
Comment (hauke) [reply package/firmware/murata-firmware/Makefile:20]: yes using the firmware blobs from murata is fine.

**Comment by thom24** [comment]
@hauke : all github checks are now ok

File: target/linux/stm32/base-files/lib/preinit/82_brcmfmac_fix_fw_names (hunk 3 above)
Comment (hauke) [reply target/linux/stm32/base-files/lib/preinit/82_brcmfmac_fix_fw_names:11]: I think showing the error message is fine if it falls back to the generic file for which you created a symlink.

File: package/utils/dtc/Makefile [hunk 9]
```diff
@@ -102,6 +103,11 @@ define Build/InstallDev
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/* $(1)/usr/lib
 endef
 
+define Host/Install
+	$(INSTALL_DIR) $(STAGING_DIR_HOSTPKG)/bin/
+	$(CP) $(MESON_HOST_BUILD_DIR)/dtc $(STAGING_DIR_HOST)/bin/
+endef
+$(eval $(call HostBuild))
```
Comment (hauke) [inline package/utils/dtc/Makefile:110]: for TF-A we just use the dtc binary from the linux kernel, see https://github.com/openwrt/openwrt/blob/main/include/trusted-firmware-a.mk#L77.

Where do you need this dtc binary?

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 10]
```diff
@@ -0,0 +1,506 @@
...
+# CONFIG_UEVENT_HELPER is not set
+CONFIG_UNCOMPRESS_INCLUDE="debug/uncompress.h"
+CONFIG_UNWINDER_ARM=y
+CONFIG_USB_SUPPORT=y
+CONFIG_USELIB=y
+CONFIG_USE_OF=y
+CONFIG_VFP=y
+CONFIG_VFPv3=y
+# CONFIG_VHOST_MENU is not set
+# CONFIG_VIRTIO_MENU is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:490]: Please remove these

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 11]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_STMMAC_ETH=y
+CONFIG_STMMAC_PLATFORM=y
+# CONFIG_STRIP_ASM_SYMS is not set
+CONFIG_SUSPEND=y
+CONFIG_SUSPEND_FREEZER=y
+CONFIG_SWPHY=y
+CONFIG_SWP_EMULATE=y
+CONFIG_SYNC_FILE=y
+# CONFIG_SYN_COOKIES is not set
+CONFIG_SYSFS_SYSCALL=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:469]: Please remove this

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 12]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_STM32_WATCHDOG=y
+CONFIG_STMMAC_ETH=y
+CONFIG_STMMAC_PLATFORM=y
+# CONFIG_STRIP_ASM_SYMS is not set
+CONFIG_SUSPEND=y
+CONFIG_SUSPEND_FREEZER=y
+CONFIG_SWPHY=y
+CONFIG_SWP_EMULATE=y
+CONFIG_SYNC_FILE=y
+# CONFIG_SYN_COOKIES is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:468]: Please activate `CONFIG_SYN_COOKIES`.

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 13]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU=y
+# CONFIG_SQUASHFS_EMBEDDED is not set
+CONFIG_SQUASHFS_FILE_CACHE=y
+# CONFIG_SQUASHFS_FILE_DIRECT is not set
+CONFIG_SQUASHFS_LZO=y
+CONFIG_SQUASHFS_ZLIB=y
+CONFIG_SRAM=y
+CONFIG_SRAM_EXEC=y
+CONFIG_STACKTRACE=y
+# CONFIG_STAGING is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:453]: Please remove this

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 14]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_SERIAL_MCTRL_GPIO=y
+CONFIG_SERIAL_STM32=y
+CONFIG_SERIAL_STM32_CONSOLE=y
+CONFIG_SERIO=y
+CONFIG_SERIO_LIBPS2=y
+CONFIG_SERIO_SERPORT=y
+CONFIG_SGL_ALLOC=y
+CONFIG_SG_POOL=y
+# CONFIG_SLAB_FREELIST_HARDENED is not set
+# CONFIG_SLAB_FREELIST_RANDOM is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:437]: Please remove these lines

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 15]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_RTC_DRV_STM32=y
+CONFIG_RTC_I2C_AND_SPI=y
+CONFIG_RTC_MC146818_LIB=y
+CONFIG_RWSEM_SPIN_ON_OWNER=y
+CONFIG_SCHED_MC=y
+# CONFIG_SCHED_STACK_END_CHECK is not set
+CONFIG_SCSI=y
+CONFIG_SCSI_COMMON=y
+# CONFIG_SECTION_MISMATCH_WARN_ONLY is not set
+# CONFIG_SECURITY_DMESG_RESTRICT is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:422]: Please remove these

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 16]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_RFS_ACCEL=y
+CONFIG_RPS=y
+CONFIG_RTC_CLASS=y
+# CONFIG_RTC_DRV_OPTEE is not set
+CONFIG_RTC_DRV_STM32=y
+CONFIG_RTC_I2C_AND_SPI=y
+CONFIG_RTC_MC146818_LIB=y
+CONFIG_RWSEM_SPIN_ON_OWNER=y
+CONFIG_SCHED_MC=y
+# CONFIG_SCHED_STACK_END_CHECK is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:418]: Please remove this

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 17]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_PREEMPT_NONE_BUILD=y
+CONFIG_PRINTK_TIME=y
+CONFIG_PTP_1588_CLOCK=y
+CONFIG_PTP_1588_CLOCK_OPTIONAL=y
+CONFIG_RANDSTRUCT_NONE=y
+CONFIG_RAS=y
+CONFIG_RATIONAL=y
+# CONFIG_RAVE_SP_CORE is not set
+CONFIG_RCU_CPU_STALL_TIMEOUT=21
+CONFIG_RCU_TRACE=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:395]: Please remove this

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 18]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_OF_FLATTREE=y
+CONFIG_OF_GPIO=y
+CONFIG_OF_IRQ=y
+CONFIG_OF_KOBJ=y
+CONFIG_OF_MDIO=y
+CONFIG_OF_OVERLAY=y
+CONFIG_OF_RESOLVE=y
+CONFIG_OID_REGISTRY=y
+CONFIG_OLD_SIGACTION=y
+CONFIG_OLD_SIGSUSPEND3=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:342]: Please remove these

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 19]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_IOMMU_SUPPORT=y
+CONFIG_IRQCHIP=y
+CONFIG_IRQSTACKS=y
+CONFIG_IRQ_DOMAIN=y
+CONFIG_IRQ_DOMAIN_HIERARCHY=y
+CONFIG_IRQ_FORCED_THREADING=y
+CONFIG_IRQ_WORK=y
+# CONFIG_ISDN is not set
+CONFIG_JBD2=y
+CONFIG_KALLSYMS=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:276]: Please remove this

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 20]
```diff
@@ -0,0 +1,506 @@
...
+CONFIG_IOMMU_IO_PGTABLE_LPAE=y
+# CONFIG_IOMMU_IO_PGTABLE_LPAE_SELFTEST is not set
+CONFIG_IOMMU_SUPPORT=y
+CONFIG_IRQCHIP=y
+CONFIG_IRQSTACKS=y
+CONFIG_IRQ_DOMAIN=y
+CONFIG_IRQ_DOMAIN_HIERARCHY=y
+CONFIG_IRQ_FORCED_THREADING=y
+CONFIG_IRQ_WORK=y
+# CONFIG_ISDN is not set
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:274]: Please remove this

File: target/linux/stm32/patches-6.6/900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch [hunk 21]
```diff
@@ -0,0 +1,51 @@
+From 63c84f846ec5b794a6c90a1c5813cdb0ae75daf6 Mon Sep 17 00:00:00 2001
+From: Thomas Richard <thomas.richard@bootlin.com>
+Date: Thu, 26 Sep 2024 16:48:52 +0200
+Subject: [PATCH] ARM: dts: stm32: add ethernet2 for STM32MP135F-DK board
+
+Signed-off-by: Thomas Richard <thomas.richard@bootlin.com>
```
Comment (hauke) [inline target/linux/stm32/patches-6.6/900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch:6]: Are you planning to add this to upstream Linux kernel too?

File: target/linux/stm32/image/Makefile [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+  IMAGE/sysupgrade.img.gz := boot-img-ext4 | sdcard-img | gzip | append-metadata
+  KERNEL := kernel-bin
+  KERNEL_NAME := zImage
+  KERNEL_IMG := zImage
+  DEVICE_DTS_DIR := $(DTS_DIR)/st
+  ENV_SIZE := 0x200000
+  DEVICE_PACKAGES := kmod-brcmfmac \
+		     murata-firmware-43430-sdio \
+		     murata-nvram-43430-sdio \
+		     kmod-mac80211 \
```
Comment (hauke) [inline target/linux/stm32/image/Makefile:47]: `kmod-mac80211` will be added indirectly not needed here.

File: target/linux/stm32/patches-6.6/900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch (hunk 21 above)
Comment (hauke) [reply target/linux/stm32/patches-6.6/900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch:6]: Which of these patches are not yet in linus mainline tree?

File: package/utils/dtc/Makefile (hunk 9 above)
Comment (thom24) [reply package/utils/dtc/Makefile:110]: I needed dtc binary for optee-os package.
And I forgot to include kernel.mk in arm-trusted-firmware-stm32.

Yes I can use dtc binary from linux kernel for optee and I added missing kernel.mk in arm-trusted-firmware-stm32 makefile, so this is not needed anymore.

File: target/linux/stm32/patches-6.6/900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch (hunk 21 above)
Comment (thom24) [reply target/linux/stm32/patches-6.6/900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch:6]: - 020-mmc-mmci-stm32-add-SDIO-in-band-interrupt-mode.patch

>  Is available upstream, but I cherry-picked the commit from the ST tree. For next iteration, I updated the patch to use the upstream commit.

- 100-ARM-dts-stm32-rtc-add-pin-to-provide-LSCO-on-stm32mp.patch
- 101-ARM-dts-stm32-rtc-add-LSCO-to-WLAN-BT-module-on-stm3.patch
- 102-ARM-dts-stm32-add-support-of-WLAN-BT-on-stm32mp135f-.patch

> These three patches were sent on mailing list, but not accepted, when I created the pull request. Now they are in the pull request sent by Alexandre Torgue for 6.13 (https://lore.kernel.org/all/92d2d6df-cc5c-488f-8ebd-550b1903db12@foss.st.com/). We can assume that they will be merged upstream during next merge window. For next iteration I updated patches to use commits from the pull request and I renamed them as 03x- (we can consider they are upstream backports).

- 700-net-stmmac-dwmac-stm32-add-support-of-phy-supply-pro.patch

> This patch is needed to enable the support of ethernet2. There is nothing available upstream.
I wrote it using ST linux 6.1 as reference.
Now ST has released 6.6 kernel, so there is a patch that I can easily backport.
For next iteration, I updated the patch to use this new one.

- 900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch

>Same than for 700-net-stmmac-dwmac-stm32-add-support-of-phy-supply-pro.patch.

For these 2 patches (700 and 900), there is nothing available upstream. But ST planned to do the job (as mentioned in the commit message https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/arch/arm/boot/dts/st/stm32mp135f-dk.dts?id=e9442f1fa4d2545dd6c0aaf7cb7a125cb04f8f2f). So I'm not planning to upstream these two patches.

Other patches are already available upstream.

Comment (hauke) [reply target/linux/stm32/patches-6.6/900-ARM-dts-stm32-add-ethernet2-for-STM32MP135F-DK-board.patch:6]: Thanks for the update. This looks fine. If they just backports we do not have to maintain the patches for ever.

File: target/linux/stm32/base-files/lib/preinit/82_brcmfmac_fix_fw_names (hunk 3 above)
Comment (thom24) [reply target/linux/stm32/base-files/lib/preinit/82_brcmfmac_fix_fw_names:11]: Ok, so I remove it

File: target/linux/stm32/image/Makefile (hunk 22 above)
Comment (thom24) [reply target/linux/stm32/image/Makefile:47]: Good question :)
Indeed this package is not needed

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 23]
```diff
@@ -0,0 +1,485 @@
...
+CONFIG_SOCK_RX_QUEUE_MAPPING=y
+CONFIG_SOC_BUS=y
+CONFIG_SOFTIRQ_ON_OWN_STACK=y
+CONFIG_SPARSE_IRQ=y
+CONFIG_SQUASHFS_DECOMP_MULTI_PERCPU=y
+# CONFIG_SQUASHFS_EMBEDDED is not set
+CONFIG_SQUASHFS_FILE_CACHE=y
+# CONFIG_SQUASHFS_FILE_DIRECT is not set
+CONFIG_SQUASHFS_LZO=y
+CONFIG_SQUASHFS_ZLIB=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:438]: Do you use squashfs? If it is not used you can also deactivate it, otherwise it is probably better to keep the default settings.

File: target/linux/stm32/stm32mp1/config-6.6 [hunk 24]
```diff
@@ -0,0 +1,485 @@
...
+CONFIG_OF_EARLY_FLATTREE=y
+CONFIG_OF_FLATTREE=y
+CONFIG_OF_GPIO=y
+CONFIG_OF_IRQ=y
+CONFIG_OF_KOBJ=y
+CONFIG_OF_MDIO=y
+CONFIG_OF_OVERLAY=y
+CONFIG_OF_RESOLVE=y
+CONFIG_OLD_SIGACTION=y
+CONFIG_OLD_SIGSUSPEND3=y
```
Comment (hauke) [inline target/linux/stm32/stm32mp1/config-6.6:339]: Please remove support for these old signals.

**Comment by hauke** [comment]
The build is currently failing in our build bots because it can not find python cryptography:
`Checking 'python3-cryptography'... failed.`
https://buildbot.openwrt.org/images/#/builders/260/builds/1
